### ISSUE TITLE
[BUG ... possibly?] Exceptions cause loss of user session when remote debugging is on.

### ISSUE THREAD

**Issue by temuri416** [phalcon/cphalcon#1624]
OK, I might be completely wrong here. But, you never know - maybe I stumbled upon some obscure real issue - so here it comes.

I use PHPEd remote debugger when developing. Every time when my application throws an exception and it is caught by PHPEd, AND I decide to terminate script execution, I lose user session.

If I let the execution continue, session is preserved.

Session is not lost when debugger is off and it does not matter if exception is left unhandled.

I have never seen anything similar in all my years dealing with remote debugging. So, my question is - could it be that Phalcon's session handler "forgets" the session variables when exception is thrown and execution is aborted from within the IDE?

I am 100% sure that PHPEd does not flush any superglobals - I've been debugging Exceptions for a long time and never experienced anything similar.

I can reliably reproduce this behaviour, if anyone from development team wants to have a closer look.

Thanks!

**Comment by temuri416** [comment]
@andresgutierrez 

Andres,

It looks like this issue has resurfaced in Phalcon 2.

Original description of the problem:

```
I use PHPEd remote debugger when developing. Every time when my application throws an exception and it is caught by PHPEd, AND I decide to terminate script execution, I lose user session.

If I let the execution continue, session is preserved.

Session is not lost when debugger is off and it does not matter if exception is left unhandled.
```

This commit shows what @sjinks did to fix it on 1.3 branch:

64a2f60c743b050795a409cda31cf19614d1b8c6

I looked at https://github.com/phalcon/cphalcon/blob/master/phalcon/session/adapter.zep and it indeed does not have a destructor.

Could you please take a look?

Thanks!

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch

**Comment by temuri416** [comment]
@andresgutierrez 

Confirmed! Thanks a lot.
